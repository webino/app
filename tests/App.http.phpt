<?php
/**
 * Webino™ (http://webino.sk)
 *
 * @noinspection PhpUnusedParameterInspection
 * @interpreter php-cgi
 * @outputmatchfile App.http.expected
 *
 * @link        https://github.com/webino/app
 * @copyright   Copyright (c) 2019 Webino, s.r.o. (http://webino.sk)
 * @author      Peter Bačinský <peter@bacinsky.sk>
 * @license     BSD-3-Clause
 */

namespace Webino;

use Tester\Assert;
use Tester\Environment;

Environment::setup();


$core = new BaseCore;
$app = $core->bootstrap();


$app->onDispatch(function (DispatchEvent $event) {
    Assert::type(HttpRequest::class, $event->getRequest());
});

$app->onHttp(function (HttpEvent $event) {
    return 'Hello Webino!';
});


$app->dispatch();
