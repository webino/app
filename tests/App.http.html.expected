%a%
Content-Type: text/html; charset=UTF-8

<html><body>Hello Webino!</body></html>
